articles/inter-vlan-routing/index.md

```diff
@@ -0,0 +1,124 @@
+
+### Inter-VLAN routing methods.
+1. **Legacy inter-vlan routing** -In this solution multiple router interfaces were used each connecting to a switch port in different vlans. The interfaces were used as default gateways and this led to some network issues as additional cabling was required when the network was to be expanded.
+
+2. **Router-on-a-stick** -In this method unlike the legacy , one physical interface port is used for routing traffic between the network segments. That is the network administrator does not require to create all the vlan interface like fa0/1 to fa0/10, all the interfaces from 1 to 10 are created in a single interface.
+3. **Layer 3 switch using switched virtual interface(SVIs)** - This is the current method of inter-vlan routing that uses layer 3/multilayer switch and switched virtual interfaces (SVI). SVIs are created for a VLANs that exists on the switch and they perform the same function for the VLANs as that of a router.
+
+Layer 3 switches are expensive hence not afforadable. In this article we are going to focus on the router-on-a-stick method for our inter-vlan routing
+
+
+### Network Configurations for Inter-VLAN communication using Router-On-Stick-Method
+
+Consider a LAN with 4 PCs,1 switch and a router connected as shown.
+
+![Local Area Network](/engineering-education/inter-vlan-routing/network.jpg)
+
+We want to configure two VLANS 10 and 20. PC0 and PC1 on VLAN10 while PC2 and PC3 are on VLAN20.
+* IP Address of PC0   -192.168.1.10
+* IP Address of PC1   -192.168.1.20
+* IP Address of PC2   -192.168.2.10
+* IP Address of PC3   -192.168.2.20
+* Default gateway for VLAN10- 192.168.1.1
+* Default gateway for VLAN10- 192.168.2.1
+
+### Step 1.
+Create two VLANS on the switch ,VLAN10 and VLAN20. Give them custom names. That is VLAN 10 -student and VLAN 20 - staff.
+```
+Switch>enable
+Switch#config terminal
+Switch(config)#vlan 10
+Switch(config-vlan)#name student
+Switch(config-vlan)#exit
+Switch(config)#vlan 20
+Switch(config-vlan)#name staff
+Switch(config-vlan)#exit
+
+```
+### Step 2.
+Assign switchports to the VLANS. Ports Fa0/1 and Fa0/2 acting as Access ports for VLAN10 while ports Fa0/3 and Fa0/4 for VLAN20.The trunk port  for carrying the traffic between the two VLANS via the router we shall use Fa0/5 port.
+```
+Switch>enable
+Switch#config terminal
+Switch(config)#int fa 0/1
+Switch(config-if)#switchport mode access
+Switch(config-if)#switchport access vlan 10
+Switch(config-if)#exit
+
+
+Switch(config)#int fa 0/2
+Switch(config-if)#switchport mode access
+Switch(config-if)#switchport access vlan 10
+Switch(config-if)#exit
+
+Switch(config)#int fa 0/3
+Switch(config-if)#switchport mode access
+Switch(config-if)#switchport access vlan 20
+Switch(config-if)#exit
+
+Switch(config)#int fa 0/4
+Switch(config-if)#switchport mode access
+Switch(config-if)#switchport access vlan 20
+Switch(config-if)#exit
```

Comment:
If possible, break down this code snippet, and explain each component separately.

Specify the objective